modules/javafx.base/src/test/java/test/javafx/util/converter/LocalDateStringConverterTest.java

```diff
@@ -48,42 +53,100 @@
+    private FormatStyle dateStyle;
+    private LocalDate validDate;
+    
     private LocalDateStringConverter converter;
     private Locale locale;
-    private FormatStyle dateStyle;
     private DateTimeFormatter formatter, parser;
-    private LocalDate validDate;
 
-    public LocalDateStringConverterTest(LocalDateStringConverter converter, Locale locale, FormatStyle dateStyle, LocalDate validDate, DateTimeFormatter formatter, DateTimeFormatter parser) {
-        this.converter = converter;
-        this.locale = locale;
+    public LocalDateStringConverterTest(LocalDateStringConverterVariant converterVariant, FormatStyle dateStyle, LocalDate validDate) {
+        this.converterVariant = converterVariant;
         this.dateStyle = dateStyle;
         this.validDate = validDate;
-        this.formatter = formatter;
-        this.parser = parser;
+
+        // initialized after Locale is established
+        this.converter = null;
+        this.locale = null;
+        this.formatter = null;
+        this.parser = null;
+    }
+
+    @BeforeClass
+    public static void setupBeforeAll() {
+        // Tests require that default locale is en_US
+        oldLocale = Locale.getDefault();
+        Locale.setDefault(Locale.US);
+
+        // DateTimeFormatter uses default locale, so we can init this after updating locale
+        aFormatter = DateTimeFormatter.ofPattern("dd MM yyyy");
+        aParser = DateTimeFormatter.ofPattern("yyyy MM dd");
+    }
+
+    @AfterClass
+    public static void teardownAfterAll() {
+        // Restore VM's old locale
+        Locale.setDefault(oldLocale);
+    }
+
+    @Before
+    public void setup() {
+        if (this.converter == null) {
+            // Locale is established now, so we can allocate objects depending on it
+            switch (this.converterVariant) {
+            case NO_PARAM:
+                this.converter = new LocalDateStringConverter();
+                this.locale = Locale.getDefault(Locale.Category.FORMAT);
+                // this.formatter and this.parser remain null
+                break;
+            case WITH_FORMATTER_PARSER:
+                this.converter = new LocalDateStringConverter(aFormatter, aParser);
+                this.locale = Locale.getDefault(Locale.Category.FORMAT);
+                this.formatter = aFormatter;
+                this.parser = aParser;
+                break;
+            case WITH_FORMAT_STYLES:
+                this.converter = new LocalDateStringConverter(FormatStyle.SHORT, Locale.UK, IsoChronology.INSTANCE);
+                this.locale = Locale.UK;
+                // this.formatter and this.parser remain null
+                break;
+            default:
+                throw new InvalidParameterException("Invalid converter variant: " + this.converterVariant.toString());
+            }
+        }
     }
 
-    @Before public void setup() {
+    @After
+    public void teardown() {
     }
```

Comment:
Minor: Since this method is empty, I think you can remove it (along with the import).

Reply:
Done